core/adapters/multiply_test.go

```diff
@@ -11,51 +12,64 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestMultiply_Unmarshal(t *testing.T) {
+	tests := []struct {
+		name    string
+		payload string
+		exp     adapters.Multiply
+	}{
+		{
+			"w/ value",
+			`{"Times": 5}`,
+			adapters.Multiply{Times: big.NewFloat(5)},
+		},
+		{
+			"w/o value",
+			`{}`,
+			adapters.Multiply{Times: nil},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var m adapters.Multiply
+			err := json.Unmarshal([]byte(tc.payload), &m)
+			require.NoError(t, err)
+			require.Equal(t, tc.exp, m)
+		})
+	}
+}
+
 func TestMultiply_Perform(t *testing.T) {
 	tests := []struct {
-		name      string
-		params    string
-		json      string
-		want      string
-		errored   bool
-		jsonError bool
+		name    string
+		Times   *big.Float
+		json    string
+		want    string
+		errored bool
 	}{
-		{"string", `{"times":100}`, `{"result":"1.23"}`, "123", false, false},
-		{"integer", `{"times":100}`, `{"result":123}`, "12300", false, false},
-		{"float", `{"times":100}`, `{"result":1.23}`, "123", false, false},
-		{"object", `{"times":100}`, `{"result":{"foo":"bar"}}`, "", true, false},
-		{"zero integer string", `{"times":0}`, `{"result":"1.23"}`, "0", false, false},
-		{"negative integer string", `{"times":-5}`, `{"result":"1.23"}`, "-6.15", false, false},
-		{"no times parameter", `{}`, `{"result":"3.14"}`, "3.14", false, false},
-
-		{"string string", `{"times":"100"}`, `{"result":"1.23"}`, "123", false, false},
-		{"string integer", `{"times":"100"}`, `{"result":123}`, "12300", false, false},
-		{"string float", `{"times":"100"}`, `{"result":1.23}`, "123", false, false},
-		{"string object", `{"times":"100"}`, `{"result":{"foo":"bar"}}`, "", true, false},
-		{"array string", `{"times":[1, 2, 3]}`, `{"result":"1.23"}`, "", false, true},
-		{"rubbish string", `{"times":"123aaa123"}`, `{"result":"1.23"}`, "", false, true},
-		{"zero string string", `{"times":"0"}`, `{"result":"1.23"}`, "0", false, false},
-		{"negative string string", `{"times":"-5"}`, `{"result":"1.23"}`, "-6.15", false, false},
+		{"by 100", big.NewFloat(100), `{"result":"1.23"}`, "123", false},
+		{"float", big.NewFloat(100), `{"result":1.23}`, "123", false},
+		{"negative", big.NewFloat(-5), `{"result":"1.23"}`, "-6.15", false},
+		{"no times parameter", nil, `{"result":"3.14"}`, "3.14", false},
+		{"object", big.NewFloat(100), `{"result":{"foo":"bar"}}`, "", true},
+		{"zero", big.NewFloat(0), `{"result":"1.23"}`, "0", false},
 	}
 
 	for _, tt := range tests {
 		test := tt
 		t.Run(test.name, func(t *testing.T) {
-			t.Parallel()
 			input := cltest.NewRunInputWithString(t, test.json)
-			adapter := adapters.Multiply{}
-			jsonErr := json.Unmarshal([]byte(test.params), &adapter)
+			adapter := adapters.Multiply{Times: test.Times}
 			result := adapter.Perform(input, nil)
 
-			if test.jsonError {
-				assert.Error(t, jsonErr)
-			} else if test.errored {
+			if test.errored {
```

Comment:
Would you mind splitting out the one error case here? Logic in tests is a bug bear of mine.